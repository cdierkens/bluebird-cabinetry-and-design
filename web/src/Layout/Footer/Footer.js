import React from "react";
import Button from "../../components/Button";
import {
  BirdIcon,
  CkdIcon,
  FacebookIcon,
  HouzzIcon,
  InstagramIcon,
  NkbaIcon,
} from "../../icons";

const Footer = () => (
  <>
    <div className="bg-gray-darker text-center text-gray-light p-5 text-sm leading-loose">
      <div className="my-6">
        <BirdIcon className="h-24 text-white" />
      </div>

      <div className="my-6">
        <div>
          <a href="tel:1-513-668-4529" className="underline">
            (513) 668‚Äê4529
          </a>
        </div>
        <div>
          <a
            href="mailto:dana@bluebirdcabinetryanddesign.com"
            className="underline text-sm sm:text-base"
          >
            dana@bluebirdcabinetryanddesign.com
          </a>
        </div>
      </div>

      <div className="my-6 flex items-center justify-center flex-wrap">
        <a href="/">
          <FacebookIcon className="h-12 text-white" />
        </a>

        <a href="/">
          <InstagramIcon className="h-12 text-white" />
        </a>

        <a href="/" className="underline">
          To The Trade Services
        </a>

        <Button to="/contact" variant="white" className="mb-5 mx-5 my-6">
          Start a Project
        </Button>
      </div>

      <div className="my-6 flex items-center justify-center flex-wrap">
        <a href="/">
          <NkbaIcon className="h-10 text-white" />
        </a>

        <a href="/">
          <CkdIcon className="h-20 text-white" />
        </a>

        <a href="/">
          <HouzzIcon className="h-10 text-white" />
        </a>
      </div>
    </div>
    <div className="bg-gray-darkest px-2 py-6 text-xs text-center text-gray-light">
      &copy; 2020 Bluebird Cabinetry &amp; Design LLC. All Rights Reserved.
    </div>
  </>
);

export default Footer;
