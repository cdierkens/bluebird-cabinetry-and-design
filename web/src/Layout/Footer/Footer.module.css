.Footer {
  @apply bg-gray-darker;
}
