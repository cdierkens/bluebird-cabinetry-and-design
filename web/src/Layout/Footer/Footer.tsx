import { Link } from "gatsby";
import React from "react";
import Button from "../../components/Button";
import {
  BirdIcon,
  CkdIcon,
  FacebookIcon,
  HouzzIcon,
  NkbaIcon,
} from "../../icons";

const Footer: React.FC<React.PropsWithChildren<unknown>> = () => (
  <>
    <div className="bg-gray-darker text-center text-gray-light p-5 text-sm leading-loose">
      <div className="my-6">
        <BirdIcon className="h-24 text-white stroke-none" />
      </div>

      <div className="my-6">
        <div>
          <a href="tel:1-513-668-4529" className="underline">
            (513) 668‚Äê4529
          </a>
        </div>
        <div>
          <a
            href="mailto:dana@bluebirdcabinetryanddesign.com"
            className="underline text-sm sm:text-base"
          >
            dana@bluebirdcabinetryanddesign.com
          </a>
        </div>
        <div>
          <span className="text-sm sm:text-base">
            Serving the{" "}
            <a
              className="underline"
              href="https://en.wikipedia.org/wiki/Cincinnati_metropolitan_area"
            >
              Greater Cincinnati Ohio Tri-State Area
            </a>
          </span>
        </div>
      </div>

      <div className="my-6 flex items-center justify-center flex-wrap">
        <a
          className="mx-2"
          href="https://www.facebook.com/kitchenandbathdesigner/"
        >
          <FacebookIcon className="h-12 text-white" />
          <span className="sr-only">Facebook</span>
        </a>

        <Link className="mx-2 underline" to="/trade-services">
          To The Trade Services
        </Link>

        <Button
          to="/contact"
          type="link"
          variant="white"
          className="mb-5 mx-5 my-6"
        >
          Contact Us
        </Button>
      </div>

      <div className="my-6 flex items-center justify-center flex-wrap">
        <a className="mx-2" href="https://nkba.org/dana-snyder">
          <NkbaIcon className="h-10 text-white" />
          <span className="sr-only">National Kitchen and Bath Association</span>
        </a>

        <CkdIcon className="mx-2 h-20 text-white" />

        <a
          className="mx-2 px-3"
          href="https://www.houzz.com/professionals/kitchen-and-bath-designers/dana-snyder-ckd-pfvwus-pf~1099041723"
        >
          <HouzzIcon className="h-10 text-white" />
          <span className="sr-only">Houzz</span>
        </a>
      </div>
    </div>
    <div className="bg-gray-darkest px-2 py-6 text-xs text-center text-gray-light">
      &copy; 2020 Bluebird Cabinetry &amp; Design LLC. All Rights Reserved.
    </div>
  </>
);

export default Footer;
