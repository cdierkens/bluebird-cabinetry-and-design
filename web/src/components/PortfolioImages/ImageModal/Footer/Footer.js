import PropTypes from "prop-types";
import React from "react";

const Footer = ({
  currentIndex,
  views,
  currentView: {
    caption,
    contractor,
    furnitureRefinishing,
    interiorDesigner,
    software,
  },
}) => {
  return (
    <div>
      <div className="inline absolute bottom-2 left-2 text-left px-2 py-1 text-sm rounded bg-gray-light">
        {contractor && <span className="block">Contractor: {contractor}</span>}
        {interiorDesigner && (
          <span className="block">Interior Designer: {interiorDesigner}</span>
        )}
        {furnitureRefinishing && (
          <span className="block">
            Furniture Refinishing: {furnitureRefinishing}
          </span>
        )}
        {software && <span className="block">Design Software: {software}</span>}
        {caption && <span className="block">{caption}</span>}
      </div>
      <div className="inline absolute bottom-2 right-2 text-right px-2 py-1 text-sm rounded bg-gray-light">
        <span>
          {currentIndex + 1} of {views.length}
        </span>
      </div>
    </div>
  );
};

Footer.propTypes = {
  currentIndex: PropTypes.number.isRequired,
  views: PropTypes.array.isRequired,
  currentView: PropTypes.object.isRequired,
};

export default Footer;
