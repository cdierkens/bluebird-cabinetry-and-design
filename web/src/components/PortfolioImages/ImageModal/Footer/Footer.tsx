import React from "react";
import { UNSAFE_ANY } from "../../../../migration.types";

const Footer: React.FC<UNSAFE_ANY> = ({
  currentIndex,
  views,
  currentView: {
    contractor,
    decorator,
    furnitureRefinishing,
    interiorDesigner,
    software,
  },
  modalProps: { isFullscreen },
}) => {
  const styles = isFullscreen
    ? "bg-black text-gray-light bg-opacity-25"
    : "bg-gray-light";

  return (
    <div>
      <div
        className={`inline absolute bottom-2 left-2 text-left px-2 py-1 text-sm rounded ${styles}`}
      >
        {contractor && <span className="block">Contractor: {contractor}</span>}

        {interiorDesigner && (
          <span className="block">Interior Designer: {interiorDesigner}</span>
        )}

        {furnitureRefinishing && (
          <span className="block">
            Furniture Refinishing: {furnitureRefinishing}
          </span>
        )}

        {software && <span className="block">Design Software: {software}</span>}

        {decorator && <span className="block">Decorator: {decorator}</span>}
      </div>
      <div
        className={`inline absolute bottom-2 right-2 text-right px-2 py-1 text-sm rounded ${styles}`}
      >
        <span>
          {currentIndex + 1} of {views.length}
        </span>
      </div>
    </div>
  );
};

export default Footer;
