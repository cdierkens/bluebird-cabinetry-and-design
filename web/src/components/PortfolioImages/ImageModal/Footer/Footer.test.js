import { render } from "@testing-library/react";
import React from "react";
import Footer from "./Footer";

it("matches snapshot", () => {
  const { asFragment } = render(
    <Footer
      currentIndex={1}
      views={[]}
      currentView={{
        caption: "",
        contractor: "",
        furnitureRefinishing: "",
        interiorDesigner: "",
        software: "",
      }}
      label="mock label"
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
