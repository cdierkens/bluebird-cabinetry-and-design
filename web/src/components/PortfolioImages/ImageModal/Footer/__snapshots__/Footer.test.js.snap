// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="inline absolute bottom-2 left-2 text-left px-2 py-1 text-sm rounded bg-gray-light"
    />
    <div
      class="inline absolute bottom-2 right-2 text-right px-2 py-1 text-sm rounded bg-gray-light"
    >
      <span>
        2 of 0
      </span>
    </div>
  </div>
</DocumentFragment>
`;
